src/models/event_query.cr

```diff
@@ -7,10 +7,15 @@ class Office365::EventQuery
   end
 
   def self.from_json(data)
-    parsed_data = JSON.parse(data).as_h["value"].as_a
-    result = parsed_data.map do |event|
-      Event.from_json(event.to_json)
+    parsed_data = JSON.parse(data).as_h
+    if parsed_data.has_key?("value")
+      array_data = parsed_data["value"].as_a
+      result = array_data.map do |event|
+        Event.from_json(event.to_json)
+      end
+      new(value: result)
+    else # error reponse
+      raise "Query: #{parsed_data.to_s} not in the correct query format"
```

Comment:
There looks to be an [`Office365::Exception`](https://github.com/PlaceOS/office365/blob/7f0f8235c20a959be7c541007da16c58e2818d8e/src/client.cr#L97-L103) that would be good to user here. The error returned from the API will also include a message which can be wrapped in this.